cpp/src/slice2cs/Gen.cpp

```diff
@@ -655,23 +722,52 @@ Slice::CsVisitor::writeDataMemberInitializers(const DataMemberList& dataMembers)
 }
 
 void
-Slice::CsVisitor::writeDocComment(const ContainedPtr& p)
+Slice::CsVisitor::writeDocComment(const ContainedPtr& p, const string& generatedType, const string& notes)
 {
     optional<DocComment> comment = DocComment::parseFrom(p, csLinkFormatter, true, true);
-    if (!comment)
+    if (comment)
     {
-        return;
+        writeDocLines(_out, "summary", comment->overview());
     }
 
-    StringList overview = comment->overview();
-    if (!overview.empty())
+    // We generate remarks only for module-level types.
+    if (dynamic_pointer_cast<Module>(p->container()))
     {
-        _out << nl << "/// <summary>";
-        writeDocLines(_out, overview);
-        _out << nl << "/// </summary>";
+        _out << nl << "/// <remarks>" << "The Slice compiler generated this " << generatedType << " from Slice "
+             << p->kindOf() << " <c>" << p->scoped() << "</c>.";
+        if (!notes.empty())
+        {
+            _out << nl << "/// " << notes;
+        }
+        _out << "</remarks>";
     }
 
-    writeSeeAlso(_out, comment->seeAlso());
+    if (comment)
+    {
+        writeSeeAlso(_out, comment->seeAlso());
+    }
+}
+
+void
+Slice::CsVisitor::writeHelperDocComment(
+    const ContainedPtr& p,
+    const string& comment,
+    const string& generatedType,
+    const string& notes)
+{
+    writeDocLine(_out, "summary", comment);
+
+    // We generate remarks only for module-level types.
+    if (dynamic_pointer_cast<Module>(p->container()))
+    {
+        _out << nl << "/// <remarks>" << "The Slice compiler generated this " << generatedType << " from Slice "
+             << p->kindOf() << " <c>" << p->scoped() << "</c>.";
```

Comment:
This might push it over the 120 limit... but it sounds weird without it.
```suggestion
        _out << nl << "/// <remarks>" << "The Slice compiler generated this " << generatedType << " from the Slice "
             << p->kindOf() << " <c>" << p->scoped() << "</c>.";
```

Reply:
I find without "the" better.